lib/logger.js

```diff
@@ -70,20 +73,35 @@ function now () {
 }
 
 function createLogger (options) {
-  if (!options.logger) {
+  // if no logger is provided, then create a default logger
+  if (!options.loggerInstance && !options.logger) {
     const logger = nullLogger
     logger.child = () => logger
     return { logger, hasLogger: false }
   }
 
-  if (validateLogger(options.logger)) {
+  if (options.logger && options.loggerInstance) {
+    throw new FST_ERR_LOGGER_AND_LOGGER_INSTANCE_PROVIDED()
+  }
+
+  // check if the logger instance has all required properties
+  if (validateLogger(options.loggerInstance)) {
     const logger = createPinoLogger({
-      logger: options.logger,
-      serializers: Object.assign({}, serializers, options.logger.serializers)
+      logger: options.loggerInstance,
+      serializers: Object.assign({}, serializers, options.loggerInstance.serializers)
     })
     return { logger, hasLogger: true }
   }
 
+  // if a logger instance is passed to logger, throw an exception
+  if (validateLogger(options.logger)) {
```

Comment:
I dont think that this is right. Here we are not checking if an pino configuration is passed, but if logger instance is passed.

Reply:
But the `validateLogger` fn checks if a valid logger object is passed. And if the user passes it to `logger` instea of `loggerInstance` it should throw an exception.

Reply:
this is connected with my remark below that we need a pino validation and not a validateLogger call here.